src/lib/music-platform/apple-music-platform.ts

```
@@ -0,0 +1,353 @@
+/**
+ * Selects tracks from a pool based on the selection mode.
+ * Randomizes within the filtered pool to provide variety.
+ */
+function selectTracksFromPool(tracks: any[], limit: number, mode: TrackSelectionMode): any[] {
+  if (tracks.length === 0) return [];
+
+  let pool: any[];
+
+  switch (mode) {
+    case 'popular':
+      const popularPoolSize = Math.max(limit, Math.ceil(tracks.length * 0.5));
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)